diesel/src/pg/expression/helper_types.rs

```diff
@@ -59,6 +59,10 @@ pub type RangeContains<Lhs, Rhs> = Grouped<
     >,
 >;
 
+/// The return type of [`lhs.overlaps_with(rhs)`](super::expression_methods::PgRangeExpressionMethods::overlaps_with)
+/// for range expressions
+#[cfg(feature = "postgres_backend")]
+pub type RangeOverlaps<Lhs, Rhs> = OverlapsWith<Lhs, Rhs>;
```

Comment:
I believe it's not necessary to have this type def as it matches the typedef of the array operators in line 41. We can just adjust the doc comment there and mention that this is also the return type of `PgRangeExperssionMethods::overlaps_with(rhs)`.

Reply:
I have removed the new type declaration and added the comment.